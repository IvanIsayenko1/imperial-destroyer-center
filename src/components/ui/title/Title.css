h1 {
  color: var(--tatooine-sand);
  text-transform: uppercase;
  letter-spacing: 2px;
}

.font-bold {
  font-weight: 700;
  text-shadow: 1px 1px 2px rgba(0, 0, 0, 0.1);
}
