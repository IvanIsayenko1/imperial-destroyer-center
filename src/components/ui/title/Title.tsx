import React from "react";
import "./Title.css";

interface TitleProps {
  title: string;
}

export const Title: React.FC<TitleProps> = ({ title }) => {
  return <h1 className="text-2xl font-bold">{title}</h1>;
};
