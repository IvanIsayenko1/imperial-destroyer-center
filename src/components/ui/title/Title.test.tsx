import { render, screen } from "@testing-library/react";
import { Title } from "./Title";

describe("Title component", () => {
  it("should render the title correctly", () => {
    render(<Title title="Test Title" />);
    const titleElement = screen.getByText("Test Title");
    expect(titleElement).toBeInTheDocument();
  });

  it("should have the correct class for styling", () => {
    render(<Title title="Test Title" />);
    const titleElement = screen.getByText("Test Title");
    expect(titleElement).toHaveClass("text-2xl", "font-bold");
  });

  it("should match the snapshot", () => {
    const { asFragment } = render(<Title title="Test Title" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
