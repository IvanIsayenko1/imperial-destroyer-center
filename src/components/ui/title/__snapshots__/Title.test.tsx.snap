// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Title component > should match the snapshot 1`] = `
<DocumentFragment>
  <h1
    class="text-2xl font-bold"
  >
    Test Title
  </h1>
</DocumentFragment>
`;
